src/ActionMonitor/Monitors/PreviewMonitor.php

```diff
@@ -9,31 +9,593 @@ class PreviewMonitor extends Monitor {
+							$node_was_updated = true;
+						}
+					}
+
+					// if the node wasn't updated, then any status we have is stale.
+					$remote_status_type = $remote_status && $node_was_updated
+						? $remote_status
+						: null;
+
+					/**
+					 * We need the above check for wether the node was updated so we
+					 * don't show stale statuses on existing nodes, but in the case that
+					 * it's a brand new draft, $node_was_updated will always be false
+					 * because at this point we're potentially getting an error on a
+					 * node that was never created. So GATSBY_PREVIEW_PROCESS_ERROR is a
+					 * special case where we always need to show the status regardless
+					 * of wether the node was updated.
+					 */
+					if ( 'GATSBY_PREVIEW_PROCESS_ERROR' === $remote_status ) {
+						$remote_status_type = $remote_status;
+					}
+
+					$status_type = 'PREVIEW_READY';
+
+					if ( ! $node_was_updated ) {
+						$status_type = 'REMOTE_NODE_NOT_YET_UPDATED';
+					}
+
+					if ( ! $found_preview_path_post_meta ) {
+						$status_type = 'NO_PREVIEW_PATH_FOUND';
+					}
+
+					$status_context = get_post_meta(
+						$post_id,
+						'_wpgatsby_node_remote_preview_status_context',
+						true
+					);
+
+					if ( $status_context === '' ) {
+						$status_context = null;
+					}
+
+					$normalized_preview_page_path =
+						$found_preview_path_post_meta !== ''
+							? $found_preview_path_post_meta
+							: null;
+
+					return [
+						'statusType'     => $status_type,
+						'statusContext'  => $status_context,
+						'remoteStatus'   => $remote_status_type,
+						'pageNode'       => [
+							'path' => $normalized_preview_page_path,
+						],
+						'modifiedLocal'  => $modified,
+						'modifiedRemote' => $gatsby_node_modified,
+					];
+				},
+			]
+		);
+
+		register_graphql_field(
+			'WPGatsby',
+			'isPreviewFrontendOnline',
+			[
+				'description' => __( 'Wether or not the Preview frontend URL is online.', 'wp-gatsby' ),
+				'type'        => 'Boolean',
+				'resolve'     => function( $root, $args, $context, $info ) {
+					if ( ! is_user_logged_in() ) {
+						return false;
+					}
+
+					$preview_url = self::get_gatsby_preview_instance_url();
+
+					$request = wp_remote_get( $preview_url );
+
+					$request_was_successful =
+						$this->was_request_successful( $request );
+
+					return $request_was_successful;
+				},
+			]
+		);
+	}
+
 	/**
-	 * Get the Gatsby Preview instance refresh webhook
+	 * Determines wether or not the current global 
+	 * PHP context is related to Gatsby Preview
 	 */
-	static function get_gatsby_preview_webhook() {
-		$preview_webhook = Preview::get_setting( 'preview_api_webhook' );
+	public static function is_gatsby_preview() {
+		$is_gatsby_preview 
+			= \is_preview()
+			|| isset( $_GET['preview_nonce'] )
+			|| isset( $_GET['gatsby_preview'] );
 
-		if (
-			! $preview_webhook ||
-			! filter_var( $preview_webhook, FILTER_VALIDATE_URL )
-		) {
-			return false;
+		return $is_gatsby_preview;
```

Comment:
@smthomas I added this helper to allow different parts to check if the current PHP request is for previews. This used to be some inline logic and didn't have the `gatsby_preview` line. That line is the magic that fixes the preview with no content changes bug where it only worked 50% of the time.